Pages/FoodPages/Index.cshtml.cs

```diff
@@ -27,27 +28,101 @@ public IndexModel(FoodKept.Data.ShopContext context, UserManager<ApplicationUser
+
+            //Sorting system
+            CurrentSort = sortOrder;
+            NameSort = string.IsNullOrEmpty(sortOrder) ? "name_desc" : "";
+            PriceSort = sortOrder == "Price" ? "price_desc" : "Price";
+            DiscountSort = sortOrder == "Discount" ? "discount_desc" : "Discount";
+            QuantitySort = sortOrder == "Quantity" ? "quantity_desc" : "Quantity";
+            FoodCategorySort = string.IsNullOrEmpty(sortOrder) ? "category_desc" : "Category";
+
+            if (searchString != null)
+            {
+                pageIndex = 1;
+            }
+            else
+            {
+                searchString = currentFilter;
+            }
+
+            CurrentFilter = searchString;
+            CurrentCategory = currentCategory;
+
             //Query for food from current user
             ApplicationUser applicationUser = await _userManager.GetUserAsync(User);
-            Food = new ModifiedList<Food>(_context.FoodData.Include(c => c.ApplicationUser).Where(c => c.ApplicationUserId == applicationUser.Id).ToList());
-           
-            //Calculate Discounts
-            CalculateCurrentPrice.CalculatePriceForFoodList(Food);
+            IQueryable<Food> foodIQ = _context.FoodData.Include(c => c.ApplicationUser).Where(c => c.ApplicationUserId == applicationUser.Id);
 
             //Filter food
-            var foods = from m in _context.FoodData
-                        select m;
-            if (!string.IsNullOrEmpty(SearchString))
+            if (!string.IsNullOrEmpty(searchString))
+            {
+                foodIQ = foodIQ.Where(s => s.FoodName.Contains(searchString));
+            }
+            else if (!string.IsNullOrEmpty(CurrentCategory))
             {
-                foods = foods.Where(s => s.FoodName.Contains(SearchString) && s.ApplicationUserId == applicationUser.Id);
-                Food = new ModifiedList<Food>(await foods.ToListAsync());
+                if (CurrentCategory != "None")
+                    foodIQ = foodIQ.Where(s => s.FoodCategory.Contains(CurrentCategory));
             }
+
+            switch (sortOrder)
+            {
+                case "name_desc":
+                    foodIQ = foodIQ.OrderByDescending(s => s.FoodName);
+                    break;
+                case "Price":
+                    foodIQ = foodIQ.OrderBy(s => s.Price);
+                    break;
+                case "price_desc":
+                    foodIQ = foodIQ.OrderByDescending(s => s.Price);
+                    break;
+                case "Discount":
+                    foodIQ = foodIQ.OrderBy(s => s.CurrentPrice.DiscountPercent);
+                    break;
+                case "discount_desc":
+                    foodIQ = foodIQ.OrderByDescending(s => s.CurrentPrice.DiscountPercent);
+                    break;
+                case "Quantity":
+                    foodIQ = foodIQ.OrderBy(s => s.Quantity);
+                    break;
+                case "quantity_desc":
```

Comment:
same issue

Reply:
I will put everything to a separate method.